# CHI-SQUARE TEST
Association between Device Type and Customer Satisfaction

Background:

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

Data Provided:
The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:

Satisfaction	Smart Thermostat	Smart Light	Total

Very Satisfied	50	70	120

Satisfied	80	100	180

Neutral	60	90	150

Unsatisfied	30	50	80

Very Unsatisfied	20	50	70

Total	240	360	600

Objective:

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

Assignment Tasks:

1. State the Hypotheses:
2. Compute the Chi-Square Statistic:
3. Determine the Critical Value:
Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)

5. Make a Decision:

Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.

Submission Guidelines:

•	Provide a detailed report of your analysis, including each step outlined in the assignment tasks in a python file.

•	Include all calculations, the Chi-Square statistic, the critical value, and your conclusion.


In [5]:
# hypothesis

#Ho=The variables "Device Type" and "Customer Satisfaction" are independent.
#H1=The variables "Device Type" and "Customer Satisfaction" are not independent.
import numpy as np
from scipy.stats import chi2_contingency

observed = np.array([[50, 70],
                     [80, 100],
                     [60, 90],
                     [30, 50],
                     [20, 50]])
alpha=0.05
# Compute Chi-Square statistic
chi2, p, dof, expected = chi2_contingency(observed)

print("Chi-Square Statistic:", chi2)
print("p_value:", p)
print("degrees of freedom:", dof)
print("expected:", expected)


Chi-Square Statistic: 5.638227513227513
p_value: 0.22784371130697179
degrees of freedom: 4
expected: [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


# by p_value calculation

In [6]:

if p < alpha:
    print("Ho rejected H1 accepted")
else:
    print("H1 rejected Ho accepted")



H1 rejected Ho accepted


by Pvalue calculation we get .. Ho accepted so they are independent so it means (independent).

# with table value result


In [7]:

from scipy.stats import chi2

# Given significance level and degrees of freedom
alpha = 0.05
#with table value result

# Determine critical value
critical_value = chi2.ppf(1 - alpha, dof)
print("Critical Value:", critical_value)


Critical Value: 9.487729036781154


chi_square table value (or)critical value=9.4877.

Chi-Square Statistic =5.6382.

clearly chi calulated value falls under accepted region.

soo Ho is accepted and H1 is rejected.

Ho=The variables "Device Type" and "Customer Satisfaction" are independent.

they are independent(no relationship ).

we have a statistical evidence to prove that there is no relationship B/W them .

and we faild to reject Ho(null hypothesis).


# HYPOTHESIS TESTING

Background:

Bombay hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation W = $1,000 + $5X, where X represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.
    
Objective:

To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

Data Provided:

•	The theoretical weekly operating cost model: W = $1,000 + $5X

•	Sample of 25 restaurants with a mean weekly cost of Rs. 3,050

•	Number of units produced in a week (X) follows a normal distribution with a mean (μ) of 600 units and a standard deviation (σ) of 25 units

Assignment Tasks:

1. State the Hypotheses statement:

2. Calculate the Test Statistic:

Use the following formula to calculate the test statistic (t):

where:

•	ˉxˉ = sample mean weekly cost (Rs. 3,050)

•	μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)

•	σ = 5*25 units

•	n = sample size (25 restaurants)

3. Determine the Critical Value:

Using the alpha level of 5% (α = 0.05), determine the critical value from the standard normal (Z) distribution table.

4. Make a Decision:

Compare the test statistic with the critical value to decide whether to reject the null hypothesis.

5. Conclusion:

Based on the decision in step 4, conclude whether there is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.

Submission Guidelines:

•	Prepare python file detailing each step of your hypothesis testing process.
    
•	Include calculations for the test statistic and the critical value.
    
•	Provide a clear conclusion based on your analysis.



# by Test Statistic calculation :

In [11]:
import numpy as np
from scipy.stats import t

# Given data
sample_mean = 3050  # Rs.
theoretical_mean = 1000 + 5 * 600  # Rs.
standard_deviation = 5 * 25  # Rs.
sample_size = 25
alpha = 0.05

# 1. State the Hypotheses statement:
# Null Hypothesis (H0): The observed weekly operating costs are consistent with the theoretical model.
# Alternative Hypothesis (H1): The observed weekly operating costs are higher than the theoretical model.

# 2. Calculate the Test Statistic:
# Using the formula: t = (sample_mean - theoretical_mean) / (standard_deviation / sqrt(sample_size))
t_cal = (sample_mean - theoretical_mean) / (standard_deviation / np.sqrt(sample_size))
print("test_statistic value:",t_cal)
# 3. Determine the Critical Value:
# Degrees of freedom for a one-sample t-test: df = sample_size - 1
df = sample_size - 1
# Critical value from t-distribution
critical_value = t.ppf(1 - alpha, df)
print("critical value:",critical_value)
# 4. Make a Decision:
# here t_cal value falls under accepted region so Ho accepted H1 rejected 
reject_null = t_cal > critical_value

# 5. Conclusion:
if reject_null:
    print("Reject the null hypothesis.")
    print("There is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.")
else:
    print("Fail to reject the null hypothesis.")
    print("There is not enough evidence to support the claim that the weekly operating costs are higher than the model suggests.")


test_statistic value: -38.0
critical value: 1.7108820799094275
Fail to reject the null hypothesis.
There is not enough evidence to support the claim that the weekly operating costs are higher than the model suggests.


here t_cal value falls under accepted region so Ho accepted H1 rejected .

# by Pvalue calculation:

In [12]:
p_value=1-stats.norm.cdf(z_cal)
print("Pvalue:",p_value)
if p_value<alpha:
    print("Ho is rejected and H1 is accepted")
else:
    print("H1 is rejected and Ho is accepted")

Pvalue: 1.0
H1 is rejected and Ho is accepted


# conclusion

Fail to reject the null hypothesis.
There is not enough evidence to support the claim that the weekly operating costs are higher than the model suggests.